connector/dialect.tdd

```diff
@@ -1,376 +1,147 @@
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='date' />
-      <argument type='date' />
-    </function>
-    <function group='date' name='MONTH' return-type='int'>
-      <formula>CAST(EXTRACT(MONTH FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='NOW' return-type='datetime'>
-      <formula>CURRENT_TIMESTAMP</formula>
-    </function>
-    <function group='date' name='QUARTER' return-type='int'>
-      <formula>CAST(EXTRACT(QUARTER FROM %1) AS INTEGER)</formula>
+    <function group='aggregate' name='MIN' return-type='datetime'>
+      <formula>MIN(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='datetime' />
     </function>
-    <function group='date' name='TODAY' return-type='date'>
-      <formula>CURRENT_DATE</formula>
-    </function>
-    <function group='date' name='WEEK' return-type='int'>
-      <formula>CAST(FLOOR((7 + EXTRACT(DOY FROM %1) - 1 + EXTRACT(DOW FROM DATE_TRUNC(&apos;YEAR&apos;, %1))) / 7) AS INTEGER)</formula>
-      <argument type='datetime' />
+    <function group='aggregate' name='SUM' return-type='real'>
+      <formula>SUM(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='real' />
     </function>
-    <function group='date' name='YEAR' return-type='int'>
-      <formula>CAST(EXTRACT(YEAR FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
+    <function group='aggregate' name='SUM' return-type='int'>
+      <formula>SUM(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='int' />
     </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>date_trunc(&apos;day&apos;, CAST(&apos;1900-01-01&apos; AS DATE) + %1 * INTERVAL &apos;1 DAY&apos;)</formula>
+    <!-- SQL-787
+    <function group='aggregate' name='STDEV' return-type='real'>
+      <formula>STDDEV_SAMP(%1)</formula>
       <argument type='real' />
     </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>CAST(%1 AS DATE)</formula>
-      <argument type='str' />
-    </function>
-    <function group='cast' name='DATE' return-type='date'>
-      <formula>CAST(%1 AS DATE)</formula>
-      <argument type='datetime' />
+    <function group='aggregate' name='STDEV' return-type='real'>
+      <formula>STDDEV_SAMP(%1)</formula>
+      <argument type='int' />
     </function>
-    <function group='cast' name='DATETIME' return-type='datetime'>
-      <formula>(CAST(&apos;1900-01-01 00:00:00&apos; AS TIMESTAMP) + %1 * INTERVAL &apos;1 DAY&apos;)</formula>
+    <function group='aggregate' name='STDEVP' return-type='real'>
+      <formula>STDDEV_POP(%1)</formula>
       <argument type='real' />
     </function>
+    <function group='aggregate' name='STDEVP' return-type='real'>
+      <formula>STDDEV_POP(%1)</formula>
+      <argument type='int' />
+    </function> -->
```

Comment:
What about the other [aggregate functions](https://docs.google.com/document/d/14DnyMSzVNRkoyib8yM62MPJIrkrG1pE0KjRYOWOUuDc/edit#heading=h.pkz5w745vwh7) we support?
- ADD_TO_ARRAY (optional distinct)
- ADD_TO_SET
- FIRST
- LAST (optional distinct)
- MERGE_OBJECTS

Should we add these to this file or no because they aren't part of the tableau list of functions?

Reply:
I tested adding a new function such as `ADD_TO_ARRAY()` and found that my original understanding of the documents was incorrect, they give a full list of [supported functions](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/FullFunctionMap.tdd), but we can add our own custom functions and it will show up in their list of available functions in Tableau.  The difference is it doesn't have a description or helper message like their supported functions do. 
I will add our functions and the different casts, currently there is only what was in the supported functions list.

Reply:
I added some of our functions such as `NULLIF`, `SUBSTRING` as they have supported argument types.  We can discuss the aggregate functions that you mentioned, I'm not sure if they can be represented in the dialect.